custom/conf/app.example.ini

```diff
@@ -1143,6 +1143,83 @@ LEVEL = Info
 ;PROJECT_BOARD_BASIC_KANBAN_TYPE = To Do, In Progress, Done
 ;PROJECT_BOARD_BUG_TRIAGE_TYPE = Needs Triage, High Priority, Low Priority, Closed
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;[project.automation]
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;ENABLED = false
+;MAX_RULES_PER_PROJECT = 25
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;[project.automation.kanban]
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;
+;; Default rules when creating an automated Kanban project.
+;; Each rule can specify a list of key/value pairs:
+;; - trigger (required)  see commands below
+;; - action (required)   see commands below
+;; - target (optional)   the rule is only active for these issue types
+;;                       one of: issue, pr, default
+;;                       can be comma-separated list (e.g. issue,pr)
+;;                       the 'default' depends on the action, but most often
+;;                       it is the same as issue,pr (notable exceptions are
+;;                       commands 'xref', 'approve' and 'assign_reviewer')
+;;                       (default value is target=default)
+;; - context (optional)  the rule is only active in this context
+;;                       one of: column:{column}, project
+;;                       where {column} is a column title
+;;                       (default value is context=project)
+;;
+;; Commands can have arguments using the command:argument syntax.
+;; They are available both as a trigger and as an action.
+;; Available commands:
+;; - move:{column}   where {column} is a column title
```

Comment:
Unfortunately, we need something else before that:
`board titles` are not unique (per project) at the moment, so we need the UNIQUE INDEX for it first.

Reply:
Yeah, having board titles here were a necessary evil I guess. I initially only supported board IDs, but then I added support for titles when I figured that we need _some_ way to configure this before being able to add automation rules via the web interface.

Then, in the end I basically only kept support for titles as to simplify the configuration interface.

While there is no guarantee of unique board titles, I still think being able to address it a board by title in the configuration is a _useful_ thing to do.  (and I don't necessarily think that having duplicate board titles within one project is a likely use-case)

Not quite sure how to proceed with this particular issue.

Reply:
I should perhaps clarify that the board/column titles are only used at project creation time and are inherently tied to the column labels in the `PROJECT_BOARD_BASIC_KANBAN_TYPE` template.  At project creation time, after the boards/columns have been created and have real IDs, the labels here are only used to lookup the IDs which are then stored in the database. Renaming the columns after creation does not "break" the rules.  The same goes for labels, which also must exist at project creation time.

While I see the value of not letting the user "shoot themselves in the foot" by supplying duplicate titles in the default configuration, I kind of see that as a separate issue that could be fixed separately instead of growing this PR even further.